specification/search/Azure.Search/models-service.tsp

```diff
@@ -0,0 +1,9176 @@
+
+    /**
+     * The description of the datasource.
+     */
+    description?: string;
+
+    /**
+     * The type of the datasource.
+     */
+    type: SearchIndexerDataSourceType;
+
+    /**
+     * Credentials for the datasource.
+     */
+    credentials: DataSourceCredentials;
+
+    /**
+     * The data container for the datasource.
+     */
+    container: SearchIndexerDataContainer;
+
+    /**
+     * The data change detection policy for the datasource.
+     */
+    dataChangeDetectionPolicy?: DataChangeDetectionPolicy;
+
+    /**
+     * The data deletion detection policy for the datasource.
+     */
+    dataDeletionDetectionPolicy?: DataDeletionDetectionPolicy;
+
+    /**
+     * The ETag of the data source.
+     */
+    @encodedName("application/json", "@odata.etag")
+    eTag?: string;
+
+    /**
+     * A description of an encryption key that you create in Azure Key Vault. This key
+     * is used to provide an additional level of encryption-at-rest for your
+     * datasource definition when you want full assurance that no one, not even
+     * Microsoft, can decrypt your data source definition. Once you have encrypted
+     * your data source definition, it will always remain encrypted. The search
+     * service will ignore attempts to set this property to null. You can change this
+     * property as needed if you want to rotate your encryption key; Your datasource
+     * definition will be unaffected. Encryption with customer-managed keys is not
+     * available for free search services, and is only available for paid services
+     * created on or after January 1, 2019.
+     */
+    encryptionKey?: SearchResourceEncryptionKey;
+}
+
+/**
+ * Represents credentials that can be used to connect to a datasource.
+ */
+model DataSourceCredentials {
+    /**
+     * The connection string for the datasource. Set to `<unchanged>` (with brackets)
+     * if you don't want the connection string updated. Set to `<redacted>` if you
+     * want to remove the connection string value from the datasource.
+     */
+    connectionString?: string;
+}
+
+/**
+ * Represents information about the entity (such as Azure SQL table or CosmosDB
+ * collection) that will be indexed.
+ */
+model SearchIndexerDataContainer {
+    /**
+     * The name of the table or view (for Azure SQL data source) or collection (for
+     * CosmosDB data source) that will be indexed.
+     */
+    name: string;
+
+    /**
+     * A query that is applied to this data container. The syntax and meaning of this
+     * parameter is datasource-specific. Not supported by Azure SQL datasources.
+     */
+    query?: string;
+}
+
+/**
+ * Base type for data change detection policies.
+ */
+@discriminator("odataType")
+model DataChangeDetectionPolicy {
+    /** The discriminator for derived types. */
+    #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "Existing"
+    @encodedName("application/json", "@odata.type")
+    odataType: string;
```

Comment:
There's no enum you can use for the discriminator here?

Reply:
Same question for other models that use `@discriminator`

Reply:
@odata.type is always a string.

It is not supposed to be specified by the user. It is baked into the model itself.

Reply:
It is used in sub-classes as discriminator. I don't think users need to read the value directly.